### PR TITLE
DAOS-8921 Coverity: Various fixes for Coverity Issues

### PR DESCRIPTION
**Pull request by dpquigl**
Coverity identified multiple issues in various C sources. This fixes CIDs:

329561, 331438, 332668, 332945, 339129, 340852

Signed-off-by: David Quigley <david.quigley@intel.com>

### CODE REVIEW

File: src/common/prop.c [hunk 1]
```diff
@@ -688,6 +688,10 @@ daos_prop_copy(daos_prop_t *prop_req, daos_prop_t *prop_reply)
 	int			 i;
 	int			 rc = 0;
 
+	if (prop_req == NULL) {
+		D_ERROR("no prop in req.\n");
+		return -DER_PROTO;
```
Comment (mchaarawi) [inline src/common/prop.c:693]:
```suggestion
		return -DER_INVAL;
```

Comment (mchaarawi) [reply src/common/prop.c:693]: (to be more consistent with other APIs)

**Review by mchaarawi — APPROVED** [review]

**Review by frostedcmos — DISMISSED** [review]
+1 for dual_iface_server. not familiar with other code to comment

**Comment by kprantis** [comment]
Looks like the only thing that failed on the last test run here is the sign-off. So perhaps ready for review?

**Review by frostedcmos — APPROVED** [review]

**Pull request merged**